src/services/items/itemFactory.ts

```diff
@@ -0,0 +1,191 @@
+        type,
+        extra: extra ?? {
+          [ItemType.ETHERPAD]: {
+            padID: faker.string.uuid(),
+            groupID: faker.string.uuid(),
+          },
+        },
+      };
+    case ItemType.SHORTCUT:
+      return {
+        type,
+        extra: extra ?? {
+          [ItemType.SHORTCUT]: {
+            target: faker.string.uuid(),
+          },
+        },
+      };
+    // bug: duplicate because of type
+    case ItemType.LOCAL_FILE:
+      return {
+        type,
+        extra: extra ?? {
+          [type]: {
+            name: faker.system.fileName(),
+            mimetype: faker.system.mimeType(),
+            path: faker.system.filePath(),
+            size: faker.number.int(),
+            altText: faker.lorem.word(),
+            content: faker.lorem.text(),
+          },
+        },
+      };
+    // bug: duplicate because of type
+    case ItemType.S3_FILE:
+      return {
+        type,
+        extra: extra ?? {
+          [type]: {
+            name: faker.system.fileName(),
+            mimetype: faker.system.mimeType(),
+            path: faker.system.filePath(),
+            size: faker.number.int(),
+            altText: faker.lorem.word(),
+            content: faker.lorem.text(),
+          },
+        },
+      };
+    case ItemType.FOLDER:
+    default:
+      return {
+        type,
+        extra: extra ?? {
+          [ItemType.FOLDER]: {
+            childrenOrder: [],
+          },
+        },
+      };
+  }
+};
+
+faker.helpers.arrayElement([
+  { lang: faker.helpers.arrayElement(['en', 'fr', 'de']) },
+  {},
+]);
+
+type ItemFactoryType<DateType = Item['createdAt']> = Pick<
+  DiscriminatedItem,
+  'id' | 'name' | 'description' | 'path' | 'settings' | 'creator' | 'extra'
+> & {
+  updatedAt: DateType;
+  createdAt: DateType;
+  // extra: ItemExtraFactory(type),
+};
+
+const adaptDate = (
+  date: Date,
+  { dateType = 'string' }: { dateType?: 'string' | 'date' } = {},
+) => {
+  if (dateType === 'string') {
+    return date.toISOString();
+  }
+  return date;
+};
+
+/**
+ *
+ * @param i partial item, can be omitted. The factory should build a correct item from given type (default to folder) and data
+ * @param options
+ * @returns
+ */
+export const ItemFactory = <DateType = string>(
+  i: Partial<DiscriminatedItem> & { parentItem?: Pick<Item, 'path'> } = {},
+  options: { dateType: 'string' | 'date' } = { dateType: 'string' },
+): ItemFactoryType<DateType> => {
+  const typeAndExtra = buildExtraAndType({
+    type: i.type ?? ItemType.FOLDER,
+    extra: i.extra,
+  });
+  const id = i.id ?? faker.string.uuid();
+  const createdAt = faker.date.anytime();
+  const updatedAt = faker.date.anytime();
+
+  const path =
+    (i.parentItem ? i.parentItem.path + '.' : '') + buildPathFromIds(id);
+
+  return {
+    id,
+    name: faker.person.fullName(),
+    description: faker.lorem.text(),
+    createdAt: adaptDate(createdAt, options) as any, // TODO need help: depend on the generic type
```

Comment:
The idea here would be for the function to get a parameter that tels how the dates should be formatted. For example: `dateFormat: 'string'` or `dateFormat: 'date'`. Based on this param, create the dates in the right format. 
The return type of the factory function should be either the one with string or with Date. The return type can be narrowed by linking the value of the argument that is passed at runtime with the output type by the means of an intermediary generic type. 

I will try to do that on my own, I let you try it too, and we can check together after.